packages/core/src/routes/domain.ts

```diff
@@ -10,18 +10,24 @@ import assertThat from '#src/utils/assert-that.js';
 import type { AuthedRouter, RouterInitArgs } from './types.js';
 
 export default function domainRoutes<T extends AuthedRouter>(
-  ...[router, { queries }]: RouterInitArgs<T>
+  ...[router, { queries, libraries }]: RouterInitArgs<T>
 ) {
   const {
     domains: { findAllDomains, findDomainById, insertDomain, deleteDomainById },
   } = queries;
+  const {
+    domains: { syncDomainStatus, addDomainToCloudflare },
+  } = libraries;
 
   router.get(
     '/domains',
     koaGuard({ response: domainResponseGuard.array(), status: 200 }),
     async (ctx, next) => {
       const domains = await findAllDomains();
-      ctx.body = domains.map((domain) => pick(domain, ...domainSelectFields));
+      const syncedDomains = await Promise.all(
+        domains.map(async (domain) => syncDomainStatus(domain))
+      );
+      ctx.body = syncedDomains.map((domain) => pick(domain, ...domainSelectFields));
```

Comment:
nit: domains =  await Promise.all(await findAllDomains().map(****))?

Reply:
inline await will trigger a lint error

Reply:
That would occur lint error "Do not access a member directly from an await expression"